pkgs/ffigen/lib/src/code_generator/objc_block.dart

```diff
@@ -405,9 +419,39 @@ $listenerName $blockingWrapper(
     }
   };
 }
-''');
-    return BindingString(
-        type: BindingStringType.objcBlock, string: s.toString());
+''';
+  }
+
+  String? _protocolTrampolineBindingString(Writer w) {
+    if (protocolTrampoline?.objCBindingsGenerated ?? true) return null;
+    protocolTrampoline!.objCBindingsGenerated = true;
+
+    final argsReceived = <String>[];
+    final argsPassed = <String>[];
+    for (var i = 0; i < params.length; ++i) {
+      final param = params[i];
+      final argName = i == 0 ? 'sel' : 'arg$i';
+      argsReceived.add(param.getNativeType(varName: argName));
+      argsPassed.add(argName);
+    }
+
+    final ret = returnType.getNativeType();
+    final argRecv = argsReceived.join(', ');
+    final argPass = argsPassed.join(', ');
+    final fnName = protocolTrampoline!.func.name;
+    final blk = w.objCLevelUniqueNamer.makeUnique('_ProtocolTrampoline');
```

Comment:
nit: Is not immediately obvious (to me) that this is "block"

Reply:
Done